css/css-grid/alignment/grid-align-baseline.html

```diff
@@ -0,0 +1,100 @@
+</div>
+
+<div class="grid">
+    <div data-expected-height="20" data-offset-y="20" class="firstRowFirstColumn alignSelfBaseline" style="height: 20px;"></div>
+    <div data-expected-height="10" data-offset-y="30" class="firstRowSecondColumn alignSelfBaseline" style="height: 10px; margin: 30px 0 10px;"></div>
+    <div data-expected-height="10" data-offset-y="30" class="firstRowThirdColumn alignSelfBaseline" style="height: 10px; margin: 20px 0 0;"></div>
+    <div data-expected-height="100" data-offset-y="0" class="firstRowFourthColumn" style="height: 100px;"></div>
+</div>
+
+<div class="grid">
+    <div id="baseline1" class="firstRowFirstColumn alignSelfBaseline">ahem</div>
+    <div id="baseline2" data-offset-y="0" class="firstRowSecondColumn alignSelfBaseline"><img src="support/1x1-blue.png" style="height: 50px;"></div>
+    <div data-expected-height="50" data-offset-y="25" class="firstRowThirdColumn alignSelfCenter"><img src="support/1x1-blue.png" style="height: 50px; vertical-align: middle"></div>
+    <div id="baseline3" data-expected-height="50" class="firstRowFourthColumn alignSelfBaseline"><img src="support/1x1-blue.png" style="height: 50px; vertical-align: middle"></div>
+    <div data-expected-height="100" data-offset-y="0" class="firstRowFifthColumn" style="height: 100px;"></div>
+</div>
+
+<div class="grid">
+    <div id="baseline4" class="firstRowFirstColumn alignSelfBaseline">ahem</div>
+    <div id="baseline5" class="firstRowSecondColumn alignSelfBaseline"><br><img src="support/1x1-blue.png" style="height: 50px;"></div>
+    <div id="baseline6" data-expected-height="50" data-offset-y="0" class="firstRowThirdColumn alignSelfBaseline"><img src="support/1x1-blue.png" style="height: 50px; vertical-align: middle"></div>
+    <div data-expected-height="100" data-offset-y="0" class="firstRowFourthColumn" style="height: 100px;"></div>
+</div>
+
+<div id="log"></div>
+
+<script>
+  var baseline1 = document.getElementById("baseline1");
+  var baseline2 = document.getElementById("baseline2");
+  var baseline3 = document.getElementById("baseline3");
+  var baseline4 = document.getElementById("baseline4");
+  var baseline5 = document.getElementById("baseline5");
+  var baseline6 = document.getElementById("baseline6");
+
+  test(() => {assert_equals(baseline1.offsetTop + baseline1.offsetHeight, baseline2.offsetTop + baseline2.offsetHeight, "baseline1 and baseline2 should have the same baseline;")},
+              "Additional Check 1");
+  test(() => {assert_less_than(baseline3.offsetTop, 25, "baseline3 should be above the center align-selfed item;")}, "Additional Check 2");
+  test(() => {assert_equals(baseline4.offsetTop, baseline5.offsetTop, "baseline4 and baseline5 should be top aligned with each other;")}, "Additional Check 3");
+  test(() => {assert_greater_than(baseline4.offsetTop, 0, "baseline4 and baseline5 should be below baseline6;")}, "Additional Check 4");
```

Comment:
This is also a typo; there is no element with Id "baseline6" in that grid. I think that it's should be better to replace that with "dominant baseline" as I commented before. An alternative would be to renumber the elements and add a new baseline6 identifier.

Reply:
With this PR I had added the id "baseline6", thinking that there was a typo - I like the idea of talking about "dominant baseline" better, though, so I'll change it here as well as inside the other test.